fapi-client/src/main/scala/com/gu/facia/api/Response.scala

```diff
@@ -15,6 +15,13 @@ case class Response[A] protected (underlying: Future[Either[ApiError, A]]) {
     }
   }
 
+  def filter(f: A => Boolean, orLeft: ApiError = FilteredOut)(implicit ex: ExecutionContext): Response[A] = flatMap { a =>
```

Comment:
This should probably be called withFilter so it can be used in for comprehensions? Although to do that canonically it would have to be implemented lazily, IIRC. Worth testing because the for-comp syntax would be the correct way of actually using this behaviour in most use-cases.